### DISCUSSION TITLE
Another question about MPI

### DISCUSSION THREAD

**Discussion by pwflanigan** [NanoComp/meep#2853] [Q&A]
I know this is something that has been discussed a lot (#2158, #2852, #2370, #2437, etc.), but I'm still confused. I'll try to ask a simple, straightforward question: I'm running MEEP on an HPC (Purdue's Anvil). When I submit a SLURM job with a line like

`mpirun -np 4 python my_code.py`

the output file has four identical copies of the standard output. Does this indicate that something is wrong with the parallelization, or is it actually splitting the problem into chunks like its supposed to? Does this indicate that MEEP was installed incorrectly, or can it be fixed by adjusting the SLURM script? Or is it not a problem at all? And looking at #2437, could adding `if mp.am_master():` be useful?

**Comment by stevengj** [comment]
Probably Meep is not installed with MPI support enabled, because `import meep` [automatically silences `stdout` except on the "master" process (0)](https://github.com/NanoComp/meep/blob/88022e8670cc03eef6a4a87c86d2acf23609aa43/python/meep.i#L1868-L1886).

You can check `meep.with_mpi()` to see if it is compiled with MPI support.  (And make sure `mpi4py` is installed too.)
